core/src/main/java/de/danoeh/antennapod/core/storage/DBReader.java

```diff
@@ -180,31 +166,20 @@ public static void loadFeedDataOfFeedItemlist(Context context,
 
     private static List<FeedItem> extractItemlistFromCursor(
             PodDBAdapter adapter, Cursor itemlistCursor) {
-        ArrayList<String> itemIds = new ArrayList<String>();
-        List<FeedItem> items = new ArrayList<FeedItem>(
-                itemlistCursor.getCount());
+        ArrayList<String> itemIds = new ArrayList<>();
+        List<FeedItem> items = new ArrayList<>(itemlistCursor.getCount());
 
         if (itemlistCursor.moveToFirst()) {
             do {
-                long imageIndex = itemlistCursor.getLong(PodDBAdapter.IDX_FI_SMALL_IMAGE);
+                int indexImage = itemlistCursor.getColumnIndex(PodDBAdapter.KEY_IMAGE);
+                long imageId = itemlistCursor.getLong(indexImage);
                 FeedImage image = null;
-                if (imageIndex != 0) {
-                    image = getFeedImage(adapter, imageIndex);
+                if (imageId != 0) {
+                    image = getFeedImage(adapter, imageId);
                 }
 
-                FeedItem item = new FeedItem(itemlistCursor.getLong(PodDBAdapter.IDX_FI_SMALL_ID),
-                        itemlistCursor.getString(PodDBAdapter.IDX_FI_SMALL_TITLE),
-                        itemlistCursor.getString(PodDBAdapter.IDX_FI_SMALL_LINK),
-                        new Date(itemlistCursor.getLong(PodDBAdapter.IDX_FI_SMALL_PUBDATE)),
-                        itemlistCursor.getString(PodDBAdapter.IDX_FI_SMALL_PAYMENT_LINK),
-                        itemlistCursor.getLong(PodDBAdapter.IDX_FI_SMALL_FEED),
-                        new FlattrStatus(itemlistCursor.getLong(PodDBAdapter.IDX_FI_SMALL_FLATTR_STATUS)),
-                        itemlistCursor.getInt(PodDBAdapter.IDX_FI_SMALL_HAS_CHAPTERS) > 0,
-                        image,
-                        itemlistCursor.getInt(PodDBAdapter.IDX_FI_SMALL_READ),
-                        itemlistCursor.getString(PodDBAdapter.IDX_FI_SMALL_ITEM_IDENTIFIER),
-                        itemlistCursor.getInt(itemlistCursor.getColumnIndex(PodDBAdapter.KEY_AUTO_DOWNLOAD)) > 0
-                        );
+                FeedItem item = FeedItem.fromCursor(itemlistCursor);
+                item.setImage(image);
```

Comment:
Image is set here.
Yeah, I could have made sure the feeditems image columns are part of the curcor and while reading the feeditem properties from the cursor also load the image properties. But this seems to be rather bad practice.